modules/tofi/hm.nix

```diff
@@ -1,34 +1,37 @@
-{
-  config,
-  lib,
-  ...
+{ config
+, lib
+, ...
 }:
 with config.stylix.fonts;
 with config.lib.stylix.colors.withHashtag; {
   options.stylix.targets.tofi.enable =
     config.lib.stylix.mkEnableTarget "Tofi" true;
 
   config = lib.mkIf config.stylix.targets.tofi.enable {
-    programs.tofi.settings = let
-      background = base00;
-      foreground = base05;
-      darkForeground = base04;
-      selection = base03;
-    in {
-      font = monospace.name;
-      font-size = toString sizes.popups;
-
-      background-color = background;
-      outline-color = darkForeground;
-      border-color = foreground;
-      text-color = foreground;
-      prompt-color = base0A;
-      prompt-background = background;
-      placeholder-color = selection;
-      input-background = background;
-      default-result-background = background;
-      selection-color = selection;
-      selection-background = background;
-    };
+    programs.tofi.settings =
+      let
+        opacity = lib.toHexString ((((builtins.ceil (config.stylix.opacity.popups * 100)) * 255) / 100));
+        background = base00 + opacity;
+        foreground = base05;
+        darkForeground = base04 + opacity;
+        selection = base03 + opacity;
+      in
+      {
+        font = monospace.name;
+        font-size = toString sizes.popups;
+        background-color = background;
+        outline-color = darkForeground;
+        border-color = foreground;
+        text-color = foreground;
+        prompt-color = base0A;
+        prompt-background = background;
+        placeholder-color = selection;
+        input-background = background;
+        default-result-background = background;
+        selection-color = selection;
+        selection-background = background;
+        border-width = lib.mkDefault 4;
+        outline-width = lib.mkDefault 2;
```

Comment:
> Stylix should aim to declare only necessary options. For example, https://github.com/danth/stylix/pull/174 removed unnecessary Stylix declarations.
>
> https://github.com/danth/stylix/pull/388#issuecomment-2126947630

IMHO, Stylix should not alter upstream default behaviours, except for colors and fonts:

```suggestion
```

Reply:
should we then go back to making the borders also have opacity too like discussed here https://github.com/danth/stylix/pull/410#discussion_r1626232736?

Reply:
@trueNAHO This was added so that we can be consistent w.r.t. border opacity as per my comment here https://github.com/danth/stylix/pull/410#discussion_r1624515882